devicetypes/qubino/qubino-flush-2-relay.src/qubino-flush-2-relay.groovy

```diff
@@ -0,0 +1,472 @@
+		if (cmd.scale == 0) {
+			eventMap.name = "energy"
+			eventMap.value = cmd.scaledMeterValue
+			eventMap.unit = "kWh"
+		} else if (cmd.scale == 2) {
+			eventMap.name = "power"
+			eventMap.value = Math.round(cmd.scaledMeterValue)
+			eventMap.unit = "W"
+		}
+	}
+	eventMap
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, ep) {
+	log.warn "Unhandled ${cmd}" + (ep ? " from endpoint $ep" : "")
+}
+
+def on() {
+	onOffCmd(0xFF)
+}
+
+def off() {
+	onOffCmd(0x00)
+}
+
+def ping() {
+	refresh()
+}
+
+def childOnOff(deviceNetworkId, value) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand onOffCmd(value, switchId)
+}
+
+private onOffCmd(value, endpoint = 1) {
+	delayBetween([
+			encap(zwave.basicV1.basicSet(value: value), endpoint),
+			encap(zwave.basicV1.basicGet(), endpoint),
+	])
+}
+
+private refreshAll(includeMeterGet = true) {
+
+	def endpoints = [1]
+
+	childDevices.each {
+		def switchId = getSwitchId(it.deviceNetworkId)
+		if (switchId != null) {
+			endpoints << switchId
+		}
+	}
+	sendHubCommand refresh(endpoints,includeMeterGet)
+}
+
+def childRefresh(deviceNetworkId, includeMeterGet = true) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) {
+		sendHubCommand refresh([switchId],includeMeterGet)
+	}
+}
+
+def refresh(endpoints = [1], includeMeterGet = true) {
+
+	def cmds = []
+
+	endpoints.each {
+		cmds << [encap(zwave.basicV1.basicGet(), it)]
+		if (includeMeterGet) {
+			cmds << encap(zwave.meterV3.meterGet(scale: 0), it)
+			cmds << encap(zwave.meterV3.meterGet(scale: 2), it)
+		}
+	}
+
+	delayBetween(cmds, 200)
+}
+
+private resetAll() {
+	childDevices.each { childReset(it.deviceNetworkId) }
+	sendHubCommand reset()
+}
+
+def childReset(deviceNetworkId) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) {
+		log.debug "Child reset switchId: ${switchId}"
+		sendHubCommand reset(switchId)
+	}
+}
+
+def reset(endpoint = 1) {
+	log.debug "Resetting endpoint: ${endpoint}"
+	delayBetween([
+			encap(zwave.meterV3.meterReset(), endpoint),
+			encap(zwave.meterV3.meterGet(scale: 0), endpoint),
+			"delay 500"
+	], 500)
+}
+
+def getSwitchId(deviceNetworkId) {
+	def split = deviceNetworkId?.split(":")
+	return (split.length > 1) ? split[1] as Integer : null
+}
+
+private encap(cmd, endpoint = null) {
+	if (cmd) {
+		if (endpoint) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+
+		if (zwaveInfo.zw.contains("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private addChildSwitches(numberOfSwitches) {
+	for (def endpoint : 2..numberOfSwitches) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = device.displayName[0..-2] + "${endpoint}"
+			addChildDevice("smartthings", "Child Metering Switch", childDni, device.getHub().getId(), [
+					completedSetup	: true,
```

Comment:
nit: missing a tab character here I think

Reply:
you mean after colon?

Reply:
before

Reply:
![qubino_code_tab](https://user-images.githubusercontent.com/41617389/84983899-0cd92b80-b13a-11ea-9529-20bb9377c2c6.png)
it's a tab

Reply:
okay. I'll blame github